addons/mail/static/src/messaging/component/notification_list/notification_list_tests.js

```diff
@@ -0,0 +1,216 @@
+odoo.define('mail.messaging.component.NotificationListTests', function (require) {
+'use strict';
+
+const NotificationList = require('mail.messaging.component.NotificationList');
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('NotificationList', {
+    beforeEach() {
+        utilsBeforeEach(this);
+
+        /**
+         * @param {Object} param0
+         * @param {string} [param0.filter='all']
+         */
+        this.createNotificationListComponent = async ({ filter = 'all' }) => {
+            NotificationList.env = this.env;
+            this.notificationList = new NotificationList(null, { filter });
+            await this.notificationList.mount(this.widget.el);
+            await afterNextRender();
+        };
+
+        this.start = async params => {
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        if (this.notificationList) {
+            this.notificationList.destroy();
+        }
+        if (this.widget) {
+            this.widget.destroy();
+        }
+        this.env = undefined;
+        delete NotificationList.env;
+    }
+});
+
+QUnit.test('notifications are ultimately ordered by date', async function (assert) {
```

Comment:
suggestion for test name: _"marked as read thread notifications are ordered by last message date"_